src/org/etools/j1939_84/controllers/part11/Part11Step13Controller.java

```diff
@@ -53,33 +87,581 @@ public class Part11Step13Controller extends StepController {
+                               // 6.2.17.22.c - Fail each PG query where any accumulator
+                               // value received is greater than FAFFh.
+                               if (spn.getRawValue() >= 0xFAFFL) {
+                                   addFailure("6.2.17.22.c - Bin value received is greater than 0xFAFFFFFF(h)"
+                                           + module.getModuleName() + " returned "
+                                           + Arrays.toString(spn.getBytes()));
+                               }
+                               // FIXME: needs to be implemented once the dataRepo is fixed
+                               // @Joe - I just need to fill in the call once I fix the dataRepo bug I found
+                               // 6.2.17.22.d - Fail all values where the corresponding value received in part 1 is
+                               // greater than the part 2 values. (where supported)
+                           });
+            }
+        }
+    }
+
+    private void testSp12730(OBDModuleInformation module) {
+        // 6.11.13.5 Actions3 for all MY2022+ Engines
+        // 6.11.13.5.a - DS request messages to ECU that indicated support in DM24 for upon request SP 12730 (GHG
+        // Tracking Lifetime Engine Run
+        // Time) for PG 64252 GHG Tracking Lifetime Array Data.
+        var ghgTrackingLifetimePackets = requestPackets(module.getSourceAddress(),
+                                                        GHG_TRACKING_LIFETIME_PGs);
+
+        GenericPacket packetForPg = haveResponseWithPg(ghgTrackingLifetimePackets, GHG_TRACKING_LIFETIME_PGs);
+            if (packetForPg == null) {
+                // 6.11.13.6.a. Fail PG query where no response was received
+                addFailure("6.11.13.6.a - No response was received from "
+                        + module.getModuleName() + "for PG "
+                        + GHG_TRACKING_LIFETIME_PGs);
+            } else {
+                packetForPg.getSpns()
+                           .stream()
+                           .forEach(spn -> {
+                               // 6.11.13.6.b. Fail PG query where any bin value received is greater than FAFFh.
+                               if (spn.getRawValue() > 0xFAFFL) {
+                                   addFailure("6.11.13.6.b - Bin value received is greater than 0xFAFFFFFFL(h)"
+                                           + module.getModuleName() + " returned "
+                                           + Arrays.toString(spn.getBytes()));
+                               }
+                               // FIXME: this needs to be implemented on the dataRepo bug is fixed
+                               // 6.11.13.6.c - Fail all values where the corresponding value received in part 2 is
+                               // greater than the part 12 value
+                               // 6.11.13.6.d - Fail if lifetime engine hours < 600 seconds
```

Comment:
6.11.13.6.d wasn't implemented.

Reply:
So, this doesn't make sense to me.  What is he is asking for is applicable to NOx binning modules.  However, this requirement exists in the GHG stuff.  So, I am confused.  At the point of this requirement we only have access to the data from PG 64252 GHG Tracking Lifetime Array Data